File: ObjectiveC/DFCI-Lib/ZoneChasseSecteur2.m
//
//  ZoneChasseSecteur2.m
//  DFCI-ObjC
//
//  Created by Patrice Trognon on 19/09/2015.
//  Copyright © 2015 Patrice Trognon. All rights reserved.
//

#import "ZoneChasseSecteur2.h"

#import "LambertII.h"
#import "Coord.h"

@implementation ZoneChasseSecteur2

-(instancetype)initWithNumeroZone:(int)numeroZone carre2Km:(Carre2Km*)carre2Km lambertOrigine:(LambertII*)lambert {
    self = [super init];
    if ( self ) {
        self.numeroZone = numeroZone ;
        self.carre2Km = carre2Km ;
        
        {
            LambertII *lambertIISO = [[LambertII alloc]initWithXLambert:lambert.x + 1000
                                                               YLambert:lambert.y + 1000];
            LatLng *latLngSO = [self latLngFromLambertII:lambertIISO];
            self.coordSO = [[Coord alloc]initWithLatLng:latLngSO lambertII:lambertIISO];
        }
        
        {
            LambertII *lambertIISE = [[LambertII alloc]initWithXLambert:lambert.x + 2000
                                                               YLambert:lambert.y + 1000];
            LatLng *latLngSE = [self latLngFromLambertII:lambertIISE];
            self.coordSE = [[Coord alloc]initWithLatLng:latLngSE lambertII:lambertIISE];
        }
        
        {
            LambertII *lambertIINE = [[LambertII alloc]initWithXLambert:lambert.x + 2000
                                                               YLambert:lambert.y + 2000];
            LatLng *latLngNE = [self latLngFromLambertII:lambertIINE];
            self.coordNE = [[Coord alloc]initWithLatLng:latLngNE lambertII:lambertIINE];
        }
        
        {
            LambertII *lambertIINO = [[LambertII alloc]initWithXLambert:lambert.x + 1000
                                                               YLambert:lambert.y + 2000 ];
            LatLng *latLngNO = [self latLngFromLambertII:lambertIINO];
            self.coordNO = [[Coord alloc]initWithLatLng:latLngNO lambertII:lambertIINO];
        }
        
        {
            LambertII *lambertIISOIntermediaire1 = [[LambertII alloc]initWithXLambert:lambert.x + 1500
                                                                             YLambert:lambert.y + 1000];
            LatLng *latLngSOIntermediaire1 = [self latLngFromLambertII:lambertIISOIntermediaire1];
            self.coordSOIntermediaire1 = [[Coord alloc]initWithLatLng:latLngSOIntermediaire1 lambertII:lambertIISOIntermediaire1];
        }
        
        {
            LambertII *lambertIISOIntermediaire2 = [[LambertII alloc]initWithXLambert:lambert.x + 1500
                                                                             YLambert:lambert.y + 1500];
            LatLng *latLngSOIntermediaire2 = [self latLngFromLambertII:lambertIISOIntermediaire2];
            self.coordSOIntermediaire2 = [[Coord alloc]initWithLatLng:latLngSOIntermediaire2 lambertII:lambertIISOIntermediaire2];
            
            self.coordCentre = [[Coord alloc]initWithLatLng:latLngSOIntermediaire2 lambertII:lambertIISOIntermediaire2];
        }
        
        {
            LambertII *lambertIISOIntermediaire3 = [[LambertII alloc]initWithXLambert:lambert.x + 1000
                                                                             YLambert:lambert.y + 1500];
            LatLng *latLngSOIntermediaire3 = [self latLngFromLambertII:lambertIISOIntermediaire3];
            self.coordSOIntermediaire3 = [[Coord alloc]initWithLatLng:latLngSOIntermediaire3 lambertII:lambertIISOIntermediaire3];
        }

    }
    return self;
}

@end

File: Java/com/ptro/dfci/ZoneChasseSecteur2.java
package com.ptro.dfci;

public class ZoneChasseSecteur2 extends CarreDFCI {

	public		int			numeroZone;
	
	public		Coord		coordSO;
	public		Coord		coordSOIntermediaire1;
	public		Coord		coordSOIntermediaire2;
	public		Coord		coordSOIntermediaire3;
	public		Coord		coordNO;
	public		Coord		coordNE;
	public		Coord		coordSE;
	public		Coord		coordCentre;
	
	public		Carre2Km	carre2Km;

	public ZoneChasseSecteur2(int numeroZone, Carre2Km carre2Km, LambertII lambert) {
		this.carre2Km = carre2Km ;
		this.numeroZone = numeroZone;
		
		{
			LambertII lambertIISO = new LambertII(lambert.getX() + 1000, 
					  							  lambert.getY() + 1000);
			LatLng latLngSO = latLngFromLambertII(lambertIISO);
			coordSO = new Coord(latLngSO, lambertIISO);
		}
		
		{
			LambertII lambertIISE = new LambertII(lambert.getX() + 2000, 
												  lambert.getY() + 1000);
			LatLng latLngSE = latLngFromLambertII(lambertIISE);
			coordSE = new Coord(latLngSE, lambertIISE);
		}

		{
			LambertII lambertIINE = new LambertII(lambert.getX() + 2000, 
												  lambert.getY() + 2000);
			LatLng latLngNE = latLngFromLambertII(lambertIINE);
			coordNE = new Coord(latLngNE, lambertIINE);
		}

		{
			LambertII lambertIINO = new LambertII(lambert.getX() + 1000, 
												  lambert.getY() + 2000);
			LatLng latLngNO = latLngFromLambertII(lambertIINO);
			coordNO = new Coord(latLngNO, lambertIINO);
		}

		{
			LambertII lambertIISE = new LambertII(lambert.getX() + 1500, 
												  lambert.getY() + 1000);
			LatLng latLngSE = latLngFromLambertII(lambertIISE);
			coordSOIntermediaire1 = new Coord(latLngSE, lambertIISE);
		}

		{
			LambertII lambertIISE = new LambertII(lambert.getX() + 1500, 
												  lambert.getY() + 1500);
			LatLng latLngSE = latLngFromLambertII(lambertIISE);
			coordSOIntermediaire2 = new Coord(latLngSE, lambertIISE);
			
			coordCentre = new Coord(latLngSE, lambertIISE);
		}

		{
			LambertII lambertIISE = new LambertII(lambert.getX() + 1000, 
												  lambert.getY() + 1500);
			LatLng latLngSE = latLngFromLambertII(lambertIISE);
			coordSOIntermediaire3 = new Coord(latLngSE, lambertIISE);
		}
	}

}
